test/common/integration/sds_integration_test.cc

```diff
@@ -0,0 +1,111 @@
+#include "envoy/config/bootstrap/v3/bootstrap.pb.h"
+#include "envoy/config/cluster/v3/cluster.pb.h"
+#include "envoy/config/core/v3/config_source.pb.h"
+#include "envoy/extensions/transport_sockets/tls/v3/cert.pb.h"
+#include "envoy/service/secret/v3/sds.pb.h"
+
+#include "test/common/integration/xds_integration_test.h"
+#include "test/config/integration/certs/clientcert_hash.h"
+#include "test/integration/ssl_utility.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace {
+
+// Hack to force linking of the service: https://github.com/google/protobuf/issues/4221.
+const envoy::service::secret::v3::SdsDummy _sds_dummy;
+
+class SdsIntegrationTest : public XdsIntegrationTest {
+public:
+  SdsIntegrationTest() : XdsIntegrationTest() {
+    skip_tag_extraction_rule_check_ = true;
+    tls_upstream_ = true;
+
+    config_helper_.addConfigModifier([this](envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
```

Comment:
does  config_helper_.configureUpstreamTls() work or is this different because sds?